compiler/ztests/join-desc.yaml

```
@@ -0,0 +1,54 @@
+script: |
+  echo "// join file desc"
+  zq -z -I file.zed
+  echo "// join pool desc"
+  export ZED_LAKE="test"
+  zed init -q 
+  zed create -q -use -orderby likes:desc people 
+  zed load -q people.zson
+  zed create -q -use -orderby flavor:desc fruits
+  zed load -q fruits.zson
+  zed query -z -I pool.zed
```

Nits:

1. Might be clearer to write output to files instead of standard output.
2. No need to quote the ZED_LAKE value.

```suggestion
  zq -z -I file.zed > file.zson
  export ZED_LAKE=test
  zed init -q 
  zed create -q -use -orderby likes:desc people 
  zed load -q people.zson
  zed create -q -use -orderby flavor:desc fruits
  zed load -q fruits.zson
  zed query -z -I pool.zed > pool.zon
```